interfaces/portal/src/app/pages/program-payment-transaction-list/program-payment-transaction-list.page.ts

```diff
@@ -75,45 +76,59 @@ export class ProgramPaymentTransactionListPageComponent {
   readonly translatableStringService = inject(TranslatableStringService);
 
   readonly table =
-    viewChild.required<QueryTableComponent<PaymentTransaction, never>>('table');
+    viewChild.required<QueryTableComponent<Transaction, never>>('table');
   readonly retryTransactionsDialog =
     viewChild.required<RetryTransactionsDialogComponent>(
       'retryTransactionsDialog',
     );
 
-  readonly contextMenuSelection = signal<PaymentTransaction | undefined>(
+  readonly contextMenuSelection = signal<Transaction | undefined>(undefined);
+
+  protected readonly paginateQuery = signal<PaginateQuery | undefined>(
     undefined,
   );
 
+  private readonly transactionsPaginateQuery = computed<PaginateQuery>(() => {
+    const paginateQuery = this.paginateQuery() ?? {};
+    return {
+      ...paginateQuery,
+      ...(paginateQuery.filter ?? {}),
+    };
+  });
+
   program = injectQuery(this.programApiService.getProgram(this.programId));
-  transactions = injectQuery(
+
+  transactionsResponse = injectQuery(
     this.paymentApiService.getPaymentTransactions({
       programId: this.programId,
       paymentId: this.paymentId,
+      paginateQuery: this.transactionsPaginateQuery,
     }),
   );
 
+  readonly transactions = computed(
+    () => this.transactionsResponse.data()?.data ?? [],
+  );
+  protected readonly totalTransactions = computed(
+    () => this.transactionsResponse.data()?.meta.totalItems ?? 0,
+  );
+
   readonly refetchPayment = signal(true);
 
   readonly columns = computed(() => {
     if (!this.program.isSuccess()) {
       return [];
     }
-    const programPaymentColumns: QueryTableColumn<PaymentTransaction>[] = [
+    const programPaymentColumns: QueryTableColumn<Transaction>[] = [
       {
         field: 'registrationProgramId',
         header: $localize`Reg. #`,
-        getCellText: (transaction) =>
-          $localize`Reg. #` + transaction.registrationProgramId.toString(),
-        getCellRouterLink: (transaction) =>
-          registrationLink({
-            programId: this.programId(),
-            registrationId: transaction.registrationId,
-          }),
-      },
-      {
-        field: 'registrationName',
-        header: $localize`Name`,
```

Comment:
I was first confused, but I get it now. This is a good thing of course to simplify/speed up the back-end query